### PR TITLE
feat(SwiftUI): Support Mistica in SwiftUI

### PR DESCRIPTION
**Pull request by jmpg93**
# Overview
Hi there! This PR is about updating mistica to support SwiftUI. The review contains over 1800 new files, but just because I've moved a lot of files and merge another repo here. 

I couldn't make more reviews because I've made 2039 changes along the way, and it was really hard to split in consistent branches.

**I highly suggest to checkout the branch and review it. Otherwise it will be really hard to review. Sorry in advice.**

## Why it is important to have a single repo?
- We only need to maintain a one repo.
- Changes are now more localised and it's easier to move from features from UIKit to SwiftUI (for example, mistica evolution).
- Single source of truth.
- Single catalog.
- Single definition of brands colors and fonts.

## Targets
The repo now has three different targets that can be build separately.
- Mistica: It contains the implementaion of UIKit
- MisticaSwiftUI: It contains the implementaion of SwiftUI
- MisticaCommon: It contains all the common thing including: colors, fonts, assets, some view utils, configuration and brand definition.

When using Mistica, the developer will only need to import `Mistica` or `MisticaSwiftUI`. `MisticaCommon` will be automatically imported, so we don't need to change our corrent WoW.

## Structure Changes
- Replace fastlane by make (I've reused the same make files of mistica-swift-ui)
- Remove Mistica project file (needed to support Carthage). Using SPM carthage is also able to fetch and build the frameworks.
- As there is no mistica project, I've been able to remove carthage dependencies (including the Cartfile).
- Update Catalog app so it can display both UIKit and SwiftUI components.
- Now there are three tests targets. They are all run in the Mistica-Package target test build.
- There is a new MisticaSwiftUI.podspec to support MisticaSwiftUI in Cocoapods.

## Code Changes
- I've made a few changes to the Catalog App to have a single app with two tabs: SwiftUI and UIKit.
- I've made a few changes to use the new colors and fonts of mistica evolution in mistica-swiftui.

## Checks
I've checked that we can work using this new structure in:
- [iphoneapp](https://github.com/Telefonica/iphoneapp/tree/mistica-upgrade) (mistica-ios added as a SPM dependency). 
- [zeus](https://github.com/Telefonica/zeus-ios/tree/mistica-upgrade) (mistica-ios added as a SPM dependency). 
- [aura-ios-container](https://github.com/Telefonica/aura-ios-container/tree/mistica-upgrade) (mistica-ios added as a Carthage dependency). 
- [smart-wifi](https://github.com/Telefonica/niji-for-home-ios/tree/mistica-upgrade) (mistica-ios added as a Cocoapods dependency). 

## Catalog demo
https://user-images.githubusercontent.com/5618731/182873000-7804f7f9-d349-4fda-97c4-2f672ed648e9.mp4

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +0,0 @@
-source "https://rubygems.org"
```
Comment (jmpg93) [inline Gemfile:1]: Not needed anymore

File: Cartfile [hunk 2]
```diff
@@ -1 +0,0 @@
-github "airbnb/lottie-ios" == 3.2.3
```
Comment (jmpg93) [inline Cartfile:1]: Not needed anymore.

File: Mistica.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -1,3161 +0,0 @@
-// !$*UTF8*$!
```
Comment (jmpg93) [inline Mistica.xcodeproj/project.pbxproj:1]: Not needed.

File: MisticaCatalog/Source/Catalog/CatalogList.swift [hunk 4]
```diff
@@ -6,16 +6,176 @@
...
-    func updateUIViewController(_ uiViewController: UICatalogViewController, context: Context) {
-        // do nothing
+    @ViewBuilder
+    func componentView(for row: CatalogRow) -> some View {
+        switch framework {
+        case .swiftUI:
+            row.swiftUIComponent
+        case .uiKit:
+            row.uiKitComponent
+        }
```
Comment (jmpg93) [inline MisticaCatalog/Source/Catalog/CatalogList.swift:48]: The same catalog view is used for both UIKit and Swif

File: MisticaCatalog/Source/Catalog/CatalogList.swift [hunk 5]
```diff
@@ -6,16 +6,176 @@
 //  Copyright © Telefonica. All rights reserved.
 //
 
+import MisticaSwiftUI
 import SwiftUI
 
-struct CatalogList: UIViewControllerRepresentable {
-    typealias UIViewControllerType = UICatalogViewController
+struct CatalogList: View {
```
Comment (jmpg93) [inline MisticaCatalog/Source/Catalog/CatalogList.swift:12]: The catalog has a few changes.

File: MisticaCatalog/Source/Catalog/MisticaSwiftUI/Components/BadgeCatalogView.swift [hunk 6]
```diff
@@ -0,0 +1,43 @@
+//
```
Comment (jmpg93) [inline MisticaCatalog/Source/Catalog/MisticaSwiftUI/Components/BadgeCatalogView.swift:1]: Real changes ends here.

File: Sources/MisticaCommon/Assets/AssetToolkit+UIImage.swift [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+//
+//  Made with ❤️ by Novum
+//
+//  Copyright © Telefonica. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+public extension UIImage {
```
Comment (jmpg93) [inline Sources/MisticaCommon/Assets/AssetToolkit+UIImage.swift:12]: Now mistica support custom assets as mistica-ios-swiftui did.

File: Sources/MisticaCommon/Fonts/FontToolkit+Font.swift [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+//  Copyright © Telefonica. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+// MARK: Font
+
+@available(iOS 13.0, *)
+public extension Font {
```
Comment (jmpg93) [inline Sources/MisticaCommon/Fonts/FontToolkit+Font.swift:15]: Extension of Font for SwiftUI.

File: Sources/MisticaCommon/Fonts/FontToolkit+UIFont.swift [hunk 9]
```diff
@@ -8,47 +8,47 @@
 
 import UIKit
 
-private var _isDynamicTypeEnabled = true
+var _isDynamicTypeEnabled = true
 
 public extension UIFont {
```
Comment (jmpg93) [inline Sources/MisticaCommon/Fonts/FontToolkit+UIFont.swift:13]: Extension of UIFont for SwiftUI.

File: Sources/MisticaCommon/Assets/AssetToolkit+Image.swift [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+//  Made with ❤️ by Novum
+//
+//  Copyright © Telefonica. All rights reserved.
+//
+
+import Foundation
+import SwiftUI
+
+@available(iOS 13.0, *)
+public extension Image {
```
Comment (jmpg93) [inline Sources/MisticaCommon/Assets/AssetToolkit+Image.swift:13]: Extension of Image that uses the UIImage exposed.

File: .github/workflows/build-mistica-catalog.yml [hunk 11]
```diff
@@ -12,46 +12,39 @@ on:
 
 jobs:
   deploy-mistica-catalog:
-    name: Deploy Mistica Catalog
+    name: Build enterprise
```
Comment (jmbrocal) [inline .github/workflows/build-mistica-catalog.yml:15]: ??

File: MisticaCatalog/SupportFiles/Info.plist [hunk 12]
```diff
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>$(PRODUCT_BUNDLE_PACKAGE_TYPE)</string>
 	<key>CFBundleShortVersionString</key>
-	<string>$(VERSION_NUMBER)</string>
+	<string>$(MARKETING_VERSION)</string>
```
Comment (cbarcom) [inline MisticaCatalog/SupportFiles/Info.plist:20]: shouldn't this be version number?

File: MisticaSwiftUI.podspec [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+  s.description      = <<-DESC
+  Mistica is an iOS framework that contains reusable UI components and utilities for the Telefonica group.
+                       DESC
+
+  s.homepage         = 'https://github.com/Telefonica/mistica-ios'
+  s.license          = { :type => 'MIT', :file => 'LICENSE' }
+  s.author           = { 'Novum' => 'cto-ios@telefonica.com' }
+  s.source           = { :git => 'https://github.com/Telefonica/mistica-ios/mistica-ios.git', :tag => s.version.to_s }
+
+  s.ios.deployment_target = '12.0'
```
Comment (cbarcom) [inline MisticaSwiftUI.podspec:22]: should this be 13.0?

Comment (cbarcom) [reply MisticaSwiftUI.podspec:22]: Sorry, I thought iOS 12 was going to be deprecated in this PR

Comment (jmpg93) [reply MisticaSwiftUI.podspec:22]: Smartwifi is still using iOS 12. So we can deal with iOS 12 for now.

**Review by cbarcom — APPROVED** [review]
amazing job!
bravo

**Review by salavert — APPROVED** [review]
gif-good-job-64

File: MisticaCatalog/SupportFiles/Info.plist (hunk 12 above)
Comment (jmpg93) [reply MisticaCatalog/SupportFiles/Info.plist:20]: Marketing version is the same as version number.
MARKETING_VERSION is the naming of the version number for the pbxproj. VERSION_NUMBERS sounds better.

image

**Pull request merged**